src/description.cpp

```diff
@@ -328,8 +376,145 @@ Description::Media::Media(const string &mline) {
 	size_t p = mline.find(' ');
 	this->type = mline.substr(0, p);
 	if (p != string::npos)
-		if (size_t q = mline.find(' ', p + 1); q != string::npos)
-			this->description = mline.substr(q + 1);
+		if (size_t q = mline.find(' ', p + 1); q != string::npos) {
+            this->description = mline.substr(q + 1, mline.find(' ', q+1)-q-2);
+        }
+}
+
+Description::Media::RTPMap& Description::Media::getFormat(int fmt) {
+    auto it = this->rtpMap.find(fmt);
+    if (it == this->rtpMap.end())
+        throw std::invalid_argument("mLineIndex is out of bounds");
+    return it->second;
+}
+
+Description::Media::RTPMap& Description::Media::getFormat(const string& fmt) {
+    for (auto &[key, val] : this->rtpMap) {
+        if (val.format == fmt)
+            return val;
+    }
+    throw std::invalid_argument("format was not found");
+}
+
+void Description::Media::removeFormat(const string &fmt) {
+    auto it = this->rtpMap.begin();
+    std::vector<int> remed;
+
+    // Remove the actual formats
+    while (it != this->rtpMap.end()) {
+        if (it->second.format == fmt) {
+            remed.emplace_back(it->first);
+            it = this->rtpMap.erase(it);
+        } else
+            it++;
+    }
+
+    // Remove any other rtpmaps that depend on the formats we just removed
+    it = this->rtpMap.begin();
+    while (it != this->rtpMap.end()) {
+        auto it2 = it->second.fmtps.begin();
+        bool rem = false;
+        while (it2 != it->second.fmtps.end()) {
+            if (it2->find("apt=") == 0) {
+                for (auto remid : remed) {
+                    if (it2->find(std::to_string(remid)) != string::npos) {
+                        std::cout << *it2 << " " << remid << std::endl;
+                        it = this->rtpMap.erase(it);
+                        rem = true;
+                        break;
+                    }
+                }
+                break;
+            }
+            it2++;
+        }
+        if (!rem)
+
+            it++;
+    }
+}
+
+void Description::Media::addVideoCodec(int payloadType, const string &codec) {
+    RTPMap map(std::to_string(payloadType) + " " + codec + "/90000");
```

Comment:
Just a suggestion, could the bitrate be a parameter here?

Reply:
The `b=AS:` parameter does indeed describe the maximum bitrate in the SDP. I just pushed a change that adds the API calls.
Thanks for the reminder on this, I am shocked that I can get 15mbps out of a browser! 

Though, REMB should be used to change bitrates dynamically, as renegotiating an SDP in-flight sounds terrifying.